redis/deprecated/redisbp/monitored_client.go

```diff
@@ -30,12 +37,47 @@ func NewMonitoredFailoverClient(name string, opt *redis.FailoverOptions) *redis.
 	return client
 }
 
+// ClusterClient extends redis cluster client with a functional Wait function
+type ClusterClient struct {
+	*redis.ClusterClient
+}
+
+// WaitArgs enclose inputs for Wait command into a struct
+type WaitArgs struct {
+	Key       string
+	NumSlaves int
+	Timeout   time.Duration
+}
```

Comment:
I'm not sure that you're really getting anything out of using an `arg` struct, all of these are required arguments from what I can tell.

Reply:
I suggested to use an struct not for optional args, but for making this a non-breaking change in the future if we need to add/remove args. The main reason is that we already have 3 args (besides context.Context) and history tells us that at that number it's very likely we'll try to add/remove args in the future 😅